lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -129,8 +156,13 @@ def join_bulk_responses(bulk_responses)
     end
 
     def bulk_send(bulk_body)
-      # Discard the URL
-      url, response = @pool.post("_bulk", nil, bulk_body)
+      params = {}
+      if use_gzip then 
+        params[:headers] = {"Content-Encoding" => "gzip"}
```

Comment:
should I make the whole header {"Content-Encoding" => "gzip"} a constant or just the string gzip from the header value? On the DEFAULT_HEADERS constant I've seen it's doing the whole header.